### PR TITLE
boards: rak: Add Rak11720 board

### PR DESCRIPTION
**Pull request by srcnert**
Added support for the Rak11720 board. Rak11720 is a WisBlock Core module with Apollo3 Blue SoC from Ambiq and a Semtech SX1262 LoRa® transceiver. For more information about this board please check: https://docs.rakwireless.com/Product-Categories/WisDuo/RAK11720-Module/Overview/#product-description

There are known issues that have been identified while communicating with SX1262 transceiver:

- Gpio interrupt handling error: ambiq_gpio_pin_interrupt_configure function is not able to configure SX1262 Dio1 interrupt pin correctly. Please check issue 73958.
- Spi communication error: Rak11720 uses a custom CS pin. Firstly, custom CS pin support is necessary.  Secondly, current spi_ambiq_xfer function is not able to communicate with SX1262 transceiver correctly. Please check issue 73959.

These issues have been documented for future solutions. Despite these limitations, the current support enables basic functionality.

Examples can be build by below command for Rak11720: west build -b rak11720 samples/hello_world

### CODE REVIEW

File: boards/rak/rak11720/rak11720.dts [hunk 1]
```diff
@@ -0,0 +1,150 @@
+/dts-v1/;
```
Comment (parthitce) [inline boards/rak/rak11720/rak11720.dts:1]: Add SPDX License identifier

Comment (srcnert) [reply boards/rak/rak11720/rak11720.dts:7]: @parthitce Added with commit b0d52b5

File: boards/rak/rak11720/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+Supported Features
+==================
+
+The RAK11720 module configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| MPU       | on-chip    | memory protection unit              |
```
Comment (parthitce) [inline boards/rak/rak11720/doc/index.rst:52]: Generally driver/component column represents the path i.e `arch/arm` for MPU and likewise for all the rest.

**Review by parthitce — DISMISSED** [review]
Fix the path in doc for `Driver/Component`, otherwise LGTM

File: boards/rak/rak11720/doc/index.rst (hunk 2 above)
Comment (srcnert) [reply boards/rak/rak11720/doc/index.rst:52]: I got this table directly from 'boards/ambiq/apollo3_evb' index.rst file. I only added lora line to table. To be honest, I do not understand what I must change.

Comment (srcnert) [reply boards/rak/rak11720/doc/index.rst]: Added with commit 279689e

**Review by parthitce — DISMISSED** [review]

**Comment by aaronyegx** [comment]
@srcnert thanks for your submission. May I know what other samples can be run for the new module/board verification? I see you are finding some peripheral issues, do you just have some local samples for LoRa testing?

**Comment by srcnert** [comment]
@aaronyegx Thanks for quick response and support. I used following example projects to test rak11720 board.

samples/hello_world
samples/ble/beacon
samples/basic/blinky
samples/basic/button
samples/sensor/sht3xd (only build, did not check it is really working)
I prepared a shtc3 sensor communication example to check I2C interface. It is working.
I checked mcuboot example. It is working on apollo3 but needs some changes on apollo3 soc definitions. I am able to update board over BLE.
If you want me to check any extra project, please inform me. 

By the way, according to previous discussions, gpio interrupt issue will be fixed at [#73890](https://github.com/zephyrproject-rtos/zephyr/pull/73890)

To check LoRa side, I used samples/subsys/lorawan/class_a example project. You can find main sx1262 driver files at drivers/lora destination. Please check #73959 for more info. I shared some logic analyzer output there. If you need more info, please inform me. Thanks.

File: boards/rak/rak11720/doc/rak11720.jpg
Comment (kartben) [inline boards/rak/rak11720/doc/rak11720.jpg:1]: please use webp format -- supports transparency and is much smaller in size, which is important to keep the size of the git repo under control :) 
see e.g. https://ae01.alicdn.com/kf/Abda6936ee57048a693dd7396f32bd57eu.png_640x640.png_.webp

File: boards/rak/rak11720/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
+.. _rak11720:
+
+RAK11720 LoRa Module
+####################
```
Comment (kartben) [inline boards/rak/rak11720/doc/index.rst:4]: to be consistent with other pages:

```suggestion
RAK11720
########
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/rak/rak11720/doc/rak11720.jpg
Comment (srcnert) [reply boards/rak/rak11720/doc/rak11720.jpg:1]: Thanks for the support. Added with commit 8d84e9b.

**Review by aaronyegx — APPROVED** [review]
LGTM.

**Review by kartben — COMMENTED** [review]
+1 for docs - thanks @srcnert!

**Review by kartben — APPROVED** [review]
+1 for docs - thanks @srcnert!

**Pull request merged**